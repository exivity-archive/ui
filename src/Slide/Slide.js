import React from 'react'
import PropTypes from 'prop-types'
import { Hidden } from 'reakit'
import AnimateHeight from 'react-animate-height'

const timeToInt = (duration) => {
  return parseInt(
    duration
      .replace('ms', '')
      .replace('s', '000')
    , 10)
}

const AnimatedHeightSlide = ({
  'aria-hidden': ariaHidden,
  children,
  visible,
  hideOnClickOutside,
  fade,
  expand,
  slide,
  duration,
  delay,
  timing,
  animated,
  ...otherProps
}) => {
  const durationInteger = animated ? timeToInt(duration) : 0
  const delayInteger = delay && timeToInt(delay)

  return <Hidden {...otherProps} visible hideOnClickOutside={hideOnClickOutside && visible}>
    <AnimateHeight
      style={{
        'WebkitMarginCollapse': 'separate'
      }}
      height={visible ? 'auto' : 0}
      easing={timing}
      duration={durationInteger}
      delay={delayInteger}>
      {children || <span />}
    </AnimateHeight>
  </Hidden>
}

const Slide = Hidden.as(AnimatedHeightSlide)

const { unmound, fade, expand, slide, ...otherHiddenPropTypes } = Hidden.propTypes

Slide.propTypes = {
  ...otherHiddenPropTypes,

  children: PropTypes.node
}

Slide.defaultProps = {
  ...Hidden.defaultProps,

  animated: true
}

Slide.displayName = 'Slide'

export default Slide
