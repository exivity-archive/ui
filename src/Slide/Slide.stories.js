import React from 'react'
import { Button, Block, Group } from 'reakit'
import { storiesOf } from '@storybook/react'
import { withPropKnobs } from '../../docs/utils/withPropKnobs'

import sample from '../../docs/examples/text.md'

import Slide from './Slide'
import Markdown from '../Markdown'

const content = <Markdown>{sample}</Markdown>

storiesOf('molecules|Slide', module)
  .add('default', () => <Slide.Container>
    {hidden => (
      <Block>
        <Group>
          <Slide.Show as={Button} {...hidden}>Show</Slide.Show>
          <Slide.Hide as={Button} {...hidden}>Hide</Slide.Hide>
          <Slide.Toggle as={Button} {...hidden}>Toggle</Slide.Toggle>
        </Group>
        {withPropKnobs(<Slide {...hidden}>{content}</Slide>)}
      </Block>
    )}
  </Slide.Container>)
