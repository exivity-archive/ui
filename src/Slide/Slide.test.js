/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Slide from './Slide'

test('renders default Slide', () => {
  const component = renderer.create(<Slide />)
  expect(component.toJSON()).toMatchSnapshot()
})
