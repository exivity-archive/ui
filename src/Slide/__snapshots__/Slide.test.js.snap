// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Slide 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  -webkit-transform-origin: calc(50% + 0px) calc(50% + 0px);
  -ms-transform-origin: calc(50% + 0px) calc(50% + 0px);
  transform-origin: calc(50% + 0px) calc(50% + 0px);
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  visibility: hidden;
  will-change: transform opacity;
}

.c2 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c2[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

<div
  aria-hidden={false}
  className="c0 c1 c2"
  hidden={false}
  onTransitionEnd={[Function]}
>
  <div
    aria-hidden={true}
    className="rah-static rah-static--height-zero"
    style={
      Object {
        "WebkitMarginCollapse": "separate",
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <div
      style={Object {}}
    >
      <span />
    </div>
  </div>
</div>
`;
